Src/HALALMock/Services/ADC/ADC.cpp

```diff
@@ -66,107 +92,61 @@ void ADC::turn_on(uint8_t id){
 		return;
 	}
 
-	uint32_t buffer_length = peripheral->init_data.channels.size();
-	if (HAL_ADC_Start_DMA(peripheral->handle, (uint32_t*) peripheral->dma_data_buffer, buffer_length) != HAL_OK) {
-		ErrorHandler("DMA - %d - of ADC - %d - did not start correctly", peripheral->init_data.dma_stream, id);
-		return;
-	}
-
-	LowPowerTimer& timer = peripheral->timer;
-	if (HAL_LPTIM_TimeOut_Start_IT(&timer.handle, timer.period, timer.period / 2) != HAL_OK) {
-		ErrorHandler("LPTIM - %d - of ADC - %d - did not start correctly", timer.name, peripheral->handle);
-		return;
-	}
 	peripheral->is_on = true;
 }
 
 float ADC::get_value(uint8_t id) {
 	Instance& instance = active_instances[id];
-	uint16_t raw = instance.peripheral->dma_data_buffer[instance.rank];
-	if(instance.peripheral->handle == &hadc3) {
-		return raw / MAX_12BIT * ADC_MAX_VOLTAGE;
+
+	EmulatedPin& emulated_pin = available_emulated_instances[instance];
+	if (emulated_pin.type != PinType::ADC) {
+		ErrorHandler("Pin %s is not configured to be used for ADC usage", emulated_pin);
+		return 0;
 	}
-	else {
-		return raw / MAX_16BIT * ADC_MAX_VOLTAGE;
+	ADCResolution resolution = instance.peripheral->init_data.emulated_resolution;
+	uint16_t raw = emulated_pin.PinData.ADC.value;
+	switch (resolution) {
+		case ADCResolution::ADC_RES_16BITS:
+			return raw / MAX_16BIT * ADC_MAX_VOLTAGE;
+		case ADCResolution::ADC_RES_14BITS:
+			return raw / MAX_14BIT * ADC_MAX_VOLTAGE;
+		case ADCResolution::ADC_RES_12BITS:
+			return raw / MAX_12BIT * ADC_MAX_VOLTAGE;
+		case ADCResolution::ADC_RES_10BITS:
+			return raw / MAX_10BIT * ADC_MAX_VOLTAGE;
```

Comment:
and then here and in the other methods do:
```suggestion
	ADCResolution resolution = static_cast<ADCResolution>(instance.peripheral->init_data.resolution);
	uint16_t raw = emulated_pin.PinData.ADC.value;
	switch (resolution) {
		case ADCResolution::ADC_RES_16BITS:
			return raw / MAX_16BIT * ADC_MAX_VOLTAGE;
		case ADCResolution::ADC_RES_14BITS:
			return raw / MAX_14BIT * ADC_MAX_VOLTAGE;
		case ADCResolution::ADC_RES_12BITS:
			return raw / MAX_12BIT * ADC_MAX_VOLTAGE;
		case ADCResolution::ADC_RES_10BITS:
			return raw / MAX_10BIT * ADC_MAX_VOLTAGE;
```